jobs/ScheduleBackupJob.js

```diff
@@ -99,22 +130,43 @@ class ScheduleBackupJob extends BaseJob {
             .deleteBackup(deleteOptions)
             .then(() => backup.backup_guid);
         }
-      }).then(deletedBackupGuids => {
-        logger.info(`Successfully deleted backup guids : ${deletedBackupGuids} - instance deleted : ${instanceDeleted}`);
+      })
+      .then(deletedBackupGuids => {
+        // Deleting transaction logs from service-container.
+        let deletedTransactionLogFilesCount = 0;
+        return backupStore.listFilesOlderThan(listOptions, transactionLogsDeleteBefore, container)
+          .map(transactionLogsFile => {
+            deletedTransactionLogFilesCount = deletedTransactionLogFilesCount + 1;
+            return backupStore.cloudProvider.remove(container, transactionLogsFile.name);
+          })
+          .then(() => {
+            const deletedObjects = {
+              deletedBackupGuids: deletedBackupGuids,
+              deletedTransactionLogFilesCount: deletedTransactionLogFilesCount
+            };
+            return deletedObjects;
+          });
+      })
+      .then(deletedObjects => {
+        logger.info(`Successfully deleted backup guids : ${deletedObjects.deletedBackupGuids} - instance deleted : ${instanceDeleted}`);
         const deleteResponse = {
-          deleted_guids: deletedBackupGuids,
+          deleted_guids: deletedObjects.deletedBackupGuids,
           job_cancelled: false,
+          deleted_transaction_log_files_count: deletedObjects.deletedTransactionLogFilesCount,
           instance_deleted: instanceDeleted
         };
         if (!instanceDeleted) {
           return deleteResponse;
         }
         logger.info(`Instance deleted. Checking if there are any more backups for :${options.instance_id}`);
         const backupStartedBefore = new Date().toISOString();
-        return backupStore
-          .listBackupFilenames(backupStartedBefore, options)
-          .then(listOfFiles => {
-            if (listOfFiles.length === 0) {
+        transactionLogsDeleteBefore = new Date().toISOString();
+        return Promise.all([
+            backupStore.listBackupFilenames(backupStartedBefore, options),
+            backupStore.listFilesOlderThan(listOptions, transactionLogsDeleteBefore, container)
+          ])
+          .spread((listOfBackups, listOfTransactionLogs) => {
+            if (listOfBackups.length === 0 && listOfTransactionLogs.length === 0) {
```

Comment:
We are **always** keeping 1 successful backup beyond retention period (even for deleted instance). So, how will this condition be satisfied?